skimage/color/colorconv.py

```diff
@@ -112,42 +124,60 @@ def convert_colorspace(arr, fromspace, tospace):
         msg = '`tospace` has to be one of {}'.format(todict.keys())
         raise ValueError(msg)
 
-    return todict[tospace](fromdict[fromspace](arr))
+    return todict[tospace](
+        fromdict[fromspace](arr, channel_axis=channel_axis),
+        channel_axis=channel_axis
+    )
 
 
-def _prepare_colorarray(arr, force_copy=False):
+def _prepare_colorarray(arr, force_copy=False, *, channel_axis=-1):
     """Check the shape of the array and convert it to
     floating point representation.
     """
     arr = np.asanyarray(arr)
 
-    if arr.shape[-1] != 3:
-        raise ValueError("Input array must have a shape == (..., 3)), "
-                         f"got {arr.shape}")
+    if arr.shape[channel_axis] != 3:
+        msg = ("the input array must have shape 3 along `channel_axis`, "
+               f"got {arr.shape}")
+        raise ValueError(msg)
 
     return dtype.img_as_float(arr, force_copy=force_copy)
 
 
-def rgba2rgb(rgba, background=(1, 1, 1)):
+def _validate_channel_axis(channel_axis, ndim):
+    if not isinstance(channel_axis, int):
+        raise TypeError("channel_axis must be an integer")
+    if channel_axis < -ndim or channel_axis >= ndim:
+        raise ValueError("channel_axis exceeds array dimensions")
```

Comment:
Did you consider raising numpy's `AxisError` in this case?

Reply:
yeah, I probably should have done that. I will update it